library/std/src/primitive_docs.rs

```diff
@@ -384,6 +384,8 @@ mod prim_char {}
 #[stable(feature = "rust1", since = "1.0.0")]
 mod prim_unit {}
 
+#[doc(alias = "ptr")]
+#[doc(alias = "pointer")]
```

Comment:
Do we need this alias? Isn't there already `#[doc(primitive = "pointer")]`?

Reply:
@pickfire please read what this does before commenting on it: https://doc.rust-lang.org/rustdoc/unstable-features.html#add-aliases-for-an-item-in-documentation-search

Reply:
Sorry, I didn't read it. Since I thought it should be straightforward such that rustdoc already thinks `primitive` as an entity or an alias, is there any reason why do we need to duplicate alias for primitive?

> Since primitive types are defined in the compiler, there's no place to attach documentation attributes. This attribute is used by the standard library to provide a way to generate documentation for primitive types.

By the way, I don't quite understand what that means. But I see that it didn't mention anything about search.

Reply:
I posted two different links, please read both.

Reply:
I already read both, I don't quite understand the second one. The first one is exactly the same thing I have in my mind, which is essentially "an alias". The second one I have seen it but I don't quite understand the documentation but I know what is it.

But my point is, why wasn't `#[doc(primitive)]` shows up in search like an alias? Is there any reason?

Reply:
@pickfire would advise not using reviews to ask for clarity on stuff in the PR. It can be misleading to triagists and other reviewers

Reply:
@Dylan-DPC  Okay, sorry about that. Can I know where should I ask?

Reply:
I share @pickfire's confusion here. If you search for [`pointer`](https://doc.rust-lang.org/nightly/std/?search=pointer) the first result is already the primitive page. Why would we also need `#[doc(alias = "pointer")]`?

If also confused and concerned about the dismissive response to @pickfire's question. As far as my interpretation is concerned it's a perfectly valid question and this is the correct place to ask it. I would also like to know the answer, maybe I can be a better response than RTFM.

Reply:
My bad, `pointer` shouldn't have been added. I think it's a bad local build that didn't make it appear. However, I think `ptr` is acceptable.

@pickfire You were absolutely right, good catch!

Reply:
```suggestion
```
Sorry, sometimes I can easily confuse others, at least it was cleared up here. :D